src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.Initialization.cs

```diff
@@ -27,11 +31,36 @@ static OpenSsl()
         }
     }
 
-    internal static partial class CryptoInitializer
+    internal static unsafe partial class CryptoInitializer
     {
-        static CryptoInitializer()
+        internal struct MemoryEntry
         {
-            if (EnsureOpenSslInitialized() != 0)
+            public char* File;
+            public int Size;
+            public int Line;
+        }
+
+        private static readonly bool DebugMemory = GetMemoryDebug("DOTNET_SYSTEM_NET_SECURITY_OPENSSL_MEMORY_DEBUG");
+        private static readonly bool ValidateMemory = GetMemoryDebug("DOTNET_SYSTEM_NET_SECURITY_OPENSSL_MEMORY_VALIDATE");
+        private static readonly IntPtr Offset = sizeof(MemoryEntry);
```

Comment:
Offset... of what?

After reading the code I understand how this works, but it may be too complicated for average dev. Would it make sense to move all memory-debug related functionality to separate file with a big comment on the top explaining how to use it and how it works (that we prefix allocations with a custom header to store allocation metadata)?